src/sorting/strand_sort.rs

```diff
@@ -0,0 +1,100 @@
+use std::collections::LinkedList;
+
+pub fn strand_sort(list: &LinkedList<i32>) -> LinkedList<i32> {
+    if list.is_empty() {
+        return LinkedList::new();
+    }
+
+    let mut ip = list.clone();
```

Comment:
Is this `clone` really needed? Could we just have some _index of active element_ in the (input) `list`?